faction.cpp

```diff
@@ -927,3 +951,37 @@ void Faction::DiscoverItem(int item, int force, int full)
 		}
 	}
 }
+
+int Faction::GetActivityCost(FactionActivity type) {
+	if (Globals->FACTION_ACTIVITY == FactionActivityRules::MARTIAL_MERGED) {
+		// do not care on particular activity type, just regions matter
+		return this->activity.size();
+	}
+
+	int count = 0;
+	for (auto &kv : this->activity) {
+		auto regionActivity = kv.second;
+
+		if (Globals->FACTION_ACTIVITY == FactionActivityRules::MARTIAL) {
+			// do not care on particular activity type, but each activity consumes one point
+			count += regionActivity.size();
+		}
+		else {
+			// standard logic, each actitivty is counted separately
+			if (regionActivity.find(type) != regionActivity.end()) {
+				count++;
+			}
+		}
+	}
+
+	return count;
+}
+
+void Faction::RecordActivity(ARegion *region, FactionActivity type) {
+	this->activity[region].insert(type);
+}
+
+bool Faction::IsActivityRecorded(ARegion *region, FactionActivity type) {
+	auto regionActivity = this->activity[region];
```

Comment:
was it pushed "as it is" or changes to auto?

Reply:
`IsActivityRecorded` is a new function.

Reply:
`auto` does not mean that variable will have a dynamic type or something. It just tells that compiler should infer type during compilation. It is the same as `var` in C# or Java.

No need to write long type names if the compiler knows them.